gallery/gallery/lib/demos/material/data_table_demo.dart

```diff
@@ -0,0 +1,554 @@
+        53,
+        4.9,
+        326,
+        2,
+        22,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithSugar(
+          localizations.dataTableRowApplePie,
+        ),
+        527,
+        26.0,
+        67,
+        7.0,
+        54,
+        12,
+        6,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowFrozenYogurt,
+        ),
+        223,
+        6.0,
+        36,
+        4.0,
+        87,
+        14,
+        1,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowIceCreamSandwich,
+        ),
+        301,
+        9.0,
+        49,
+        4.3,
+        129,
+        8,
+        1,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowEclair,
+        ),
+        326,
+        16.0,
+        36,
+        6.0,
+        337,
+        6,
+        7,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowCupcake,
+        ),
+        369,
+        3.7,
+        79,
+        4.3,
+        413,
+        3,
+        8,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowGingerbread,
+        ),
+        420,
+        16.0,
+        61,
+        3.9,
+        327,
+        7,
+        16,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowJellyBean,
+        ),
+        439,
+        0.0,
+        106,
+        0.0,
+        50,
+        0,
+        0,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowLollipop,
+        ),
+        456,
+        0.2,
+        110,
+        0.0,
+        38,
+        0,
+        2,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowHoneycomb,
+        ),
+        472,
+        3.2,
+        99,
+        6.5,
+        562,
+        0,
+        45,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowDonut,
+        ),
+        516,
+        25.0,
+        63,
+        4.9,
+        326,
+        2,
+        22,
+      ),
+      _Dessert(
+        localizations.dataTableRowWithHoney(
+          localizations.dataTableRowApplePie,
+        ),
+        582,
+        26.0,
+        77,
+        7.0,
+        54,
+        12,
+        6,
+      ),
+    ];
+  }
+
+  final BuildContext context;
+  List<_Dessert> _desserts;
+
+  void _sort<T>(Comparable<T> getField(_Dessert d), bool ascending) {
+    _desserts.sort((a, b) {
+      if (!ascending) {
+        final _Dessert c = a;
+        a = b;
+        b = c;
+      }
+      final Comparable<T> aValue = getField(a);
+      final Comparable<T> bValue = getField(b);
+      return Comparable.compare(aValue, bValue);
```

Comment:
I think this block might be more readable if as:
```
final Comparable<T> aValue = getField(a);
final Comparable<T> bValue = getField(b);
return ascending ? Comparable.compare(aValue, bValue) : Comparable.compare(bValue, aValue);
```

Reply:
done